### PR TITLE
keep screen off between runs

### PR DESCRIPTION
**Pull request by yjbanov**
Fixes https://github.com/flutter/flutter/issues/4314

/cc @devoncarew

### CODE REVIEW

File: lib/src/adb.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  /// Sends `KEYCODE_POWER` (26), which causes the device to toggle its mode
+  /// between awake and asleep.
+  Future<Null> togglePower() async {
+    await shellExec('input', const ['keyevent', '26']);
+  }
+
+  /// Unlocks the device by sending `KEYCODE_MENU` (82).
+  ///
+  /// This only works when the devices doesn't have a secure unlock pattern.
```
Comment (devoncarew) [inline lib/src/adb.dart]: nit: grammar - device doesn't

File: lib/src/adb.dart [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import 'utils.dart';
+
+typedef Adb AdbGetter();
+
+/// Get an instance of [Adb].
+///
+/// See [realAdb] for signature. This can be overwritten for testing.
+AdbGetter adb = realAdb;
+
+Adb realAdb() {
```
Comment (devoncarew) [inline lib/src/adb.dart]: `createRealAdb()`?

File: lib/src/adb.dart [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  }
+
+  /// Whether the device is asleep.
+  Future<bool> isAsleep() async {
+    return await _getWakefulness() == 'Asleep';
+  }
+
+  /// Wake up the device if it is not awake using [togglePower].
+  Future<Null> wakeUp() async {
+    if (!(await isAwake())) {
```
Comment (devoncarew) [inline lib/src/adb.dart]: nit: unneeded braces, and in `sendToSleep` below as well

**Comment by devoncarew** [comment]
lgtm

I assume the device with the screen on is drawing power faster than it can charge?

File: lib/src/adb.dart (hunk 1 above)
Comment (yjbanov) [inline lib/src/adb.dart]: Done

File: lib/src/adb.dart (hunk 2 above)
Comment (yjbanov) [inline lib/src/adb.dart]: Done

File: lib/src/adb.dart (hunk 3 above)
Comment (yjbanov) [inline lib/src/adb.dart]: Done

**Comment by yjbanov** [comment]
> I assume the device with the screen on is drawing power faster than it can charge?

I did see the battery at around 60% and occasionally the device would power down completely, although I never caught the exact shutdown moment and so I'm not 100% sure it powers down due to battery.

**Pull request merged**